hiclass/metrics.py

```diff
@@ -248,3 +252,526 @@ def _compute_macro(y_true: np.ndarray, y_pred: np.ndarray, _micro_function):
         sample_score = _micro_function(np.array([ground_truth]), np.array([prediction]))
         overall_sum = overall_sum + sample_score
     return overall_sum / len(y_true)
+
+
+def _prepare_data(
+    classifier: HierarchicalClassifier,
+    y_true: np.ndarray,
+    y_prob: np.ndarray,
+    level: int,
+    y_pred: np.ndarray = None,
+):
+    classifier_classes = np.array(classifier.classes_[level]).astype("str")
+    y_true = make_leveled(y_true)
+    y_true = classifier._disambiguate(y_true)
```

Comment:
I am not sure I follow why make_leveled and _disambiguate need to be called here.

Reply:
Training and calibration samples are transformed using these methods in the HierarchicalClassifier class. I apply the same transformations for the test samples before calculating the metrics to get the classes for each level seperately (instead of a list of labels for each data point).